SPIREX-181 Pull changes from Spire Main Repository

willallves opened the pull request.

glaucimar committed:
Update readme (#2)

* README updates

* Update README.md

glaucimar committed:
Update README.md

willallves committed:
SPIREX-63 Change K8s wl to validate signature subject as a selector (#6)

SPIREX-63 Change K8s wl to validate signature subject as a selector (#6)

marcusvtms committed:
tests: Integration tests for sigstore workload attestor

marcusvtms committed:
Spirex 114 implement final solution for dockerhub pull rate limit in integration tests (#11)

rodrigolc committed:
feat: refactored sigstore interactions to a separate package

rodrigolc committed:
fix: bumping cosign to v1.4

rodrigolc committed:
fix: go.mod and go.sum after make

rodrigolc committed:
quick fixes to subject test

rodrigolc committed:
misc: improved logging

rodrigolc committed:
fix: fixed coments breaking linter tests

rodrigolc committed:
misc:fixing linting again

otaviolcs3 committed:
Implement unit tests for Attestor (#14)

* Refactoring sigstore requests from k8s to sigstore package

* Change call from cosign.verify for a helper function in order to enable Mocking

* First set of tests

* Fixing exception cases

* Linting issue

* reverting the debug change

Co-authored-by: Thiago Jamir <tjamir@gmail.com>

rodrigolc committed:
Merge pull request #16 from rodrigolc/test-sigstore-mock

improved testing and coverage with sigstore package mock

rodrigolc committed:
Image skiplist support (#18)

* feat: Refactoring of sigstore package, added digest validation, added skip list

* fix: added empty signature checking on getSignatureSubject

* updated tests after refactoring

* feat: added skiplist feature, adjusted after refactoring

* feat: fixed tests after refactoring

* fixing linting

* tests: added AddSkipImage tests

* tests: improved testing for sigstore.go

* feat: added signature-verified selector for verified image signatures

* tests: implemented tests covering code added to Attest
tests: added tests for skip list config

* fix: fixing linting issues

* fix another linting issue

* feat: the skip list in sigstore.go doesn't store selectors anymore,
feat: the k8s attestor now adds the signature-verified:true selector to images in the skip list, and doesn't add any other sigstore related selectors beyond that

* docs: updated agent_full.conf to reflect changes in the attestor

* misc: removed old comment

* misc: renamed SkipImage to ShouldSkipImage

* misc: renamed SkipImage in comments too

* misc: removing unused functions

willallves committed:
SPIREX-102 Implement workload Registrar for Sigstore on Webhook (#13)

matheusdefarias committed:
Merge pull request #20 from matheusdefarias/dev-signature-content

[SPIREX-25] Add signature content selector

joaoguazzelli committed:
Add signature-content and signature-verified selectors documentation (#23)

rodrigolc committed:
Implementing Subjects Allowed List (#19)

* updated tests after refactoring

* feat: added allow_list support to the k8s/sigstore package

* feat: added allowlist support for the k8s.go plugin

* docs: added config docs for allowlist options

* Fixing linting issues

* misc: addressing comments on PR

* misc: changed emails from hpe.com to example.com

* tests: improved code coverage

* misc: fixing linting complains

* misc: fixing linting, again

willallves committed:
SPIREX-126 Implement workload Registrar for Sigstore on CRD (#24)

rodrigolc committed:
Dev refactor k8s interfacing code (#21)

* misc: changes to skipImage

* misc: moved main logic to `sigstore.go`

* misc: fixed tests for k8s.go

* tests: added tests for AttestContainerSignatures

* fix: setting rekorURL on config

* refactor!: moved URL parsing and error handling to its own function

* tests: updated tests to cover sigstore configuration in k8s.go

* fix: fixing integration tests, empty url means use default

rodrigolc committed:
SPIREX-24 - Implement "log id" and "integrated date" selector (#25)

* feat: added selectors for logID and integratedTime

* misc: removing last reference to hpe

* misc: still had a hpe reference.

* tests: added integration test for new selectors

* tests:fixed app name on entry creation on new test

* ci: fixing k8s-sigstore integration test for extra selectors

rodrigolc committed:
tests: added new integration tests for registrar and skip/allow lists (#27)

* tests: added new integration tests for registrar and skip/allow lists

matheusdefarias committed:
dev: change k8s workload to check all pods image

* refactor: adjusted containerID place

* fix: lint adjustment

* fix: adjusted sigstore test file

* fix: adjusted sigstore test file

* fix: fixed k8s-sigstore integration tests

* refactor: removed some comments and renamed k8s-sigstore-e2e folder to .k8s-sigstore-e2e

* refactor: removed unnecessary line of code

Co-authored-by: Matheus Santos <mfcs@cesar.org.br>

willallves committed:
SPIREX-86 Implement a cache strategy for storing signature (#28)

joaoguazzelli committed:
test: Create integration test for multiple containers in a pod

rodrigolc committed:
Sigstore crd (#30)

* adding missing logic on CRD k8s-workload-registrar

* fixed tests

* fix: removed sigstore selector in node registration entry

* fix: removed old log

* fix: lint and unit tests

* fix: fixed flooding of registrar logs.
fix: improved coverage to include sigstore selector inclusion

* tests: modified tests to account for CheckSignatureEnabled

* fix: addressing lint complaints

* removing unneeded field

* renamed last mentions of "signature-verified:true"

* fix: integration tests had a double "$"

willallves committed:
merge spire/main into hpe

Signed-off-by: Willian Alves <wiilliian.alves@gmail.com>

matheusdefarias committed:
refactor: removed containerid from sigstore-validation selector and included again k8s-sigstore-e2e tests (#34)

Co-authored-by: Matheus Santos <mfcs@cesar.org.br>

willallves committed:
fixbug: unittests windows

rodrigolc commented on doc/plugin_agent_workloadattestor_k8s.md:
```diff
@@ -59,7 +59,8 @@ server name validation against the kubelet certificate.
 | k8s:pod-init-image-count | The number of init container images in workload's pod |
-
+| k8s:signature-content    | The value of the signature itself in a hash|
+| k8s:sigstore-validation   | The confirmation if the signature is valid, has value of "passed" (eg. "k8s:sigstore-validation:passed") |
```
missing other selector values for sigstore

willallves replied:
> missing other selector values for sigstore

added

tjamir replied:
In current version, the subject and log id are still missing.

rodrigolc commented on test/integration/suites/k8s/conf/kind-config.yaml:
```diff
@@ -18,3 +18,4 @@ nodes:
   extraMounts:
   - containerPath: /etc/kubernetes/pki/admctrl
     hostPath: CONFDIR/admctrl
+    
```
double line ending on file ending

rodrigolc reviewed (changes requested):
minor changes

rodrigolc committed:
Fix sigstore cache (#35)

* fix: resolving a cache bug (sigstore tries to fetch signatures even if cached

* fixing lint issues

* refactor: changing how signature are cached

* tests: fixed mock returning old selector string

Co-authored-by: Thiago Jamir <tjamir@gmail.com>
Conflicts:
	pkg/agent/plugin/workloadattestor/k8s/k8s.go
	pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore.go
	pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go

tjamir commented on README.md:
```diff
@@ -71,3 +71,9 @@ A third party security firm ([Cure53](https://cure53.de/)) completed a security
+
+This repo is a fork of the [SPIRE repository](https://github.com/spiffe/spire) and enhances Kubernetes workload attestor to validate sigstore signatures.
+
+The RFC is available [here](https://docs.google.com/document/d/1YVuu7HMHnp8nx3sCPx7R2lCfjjno363s4oiPlI6axF4/edit#heading=h.ttn87ugq19sb) for reference.
```
This is a PR. I think shouldn't have any metion that is a fork from Spire.
Instead, we should add some content to the docs descrbing the Attestor change.

tjamir commented on doc/plugin_agent_workloadattestor_k8s.md:
```diff
@@ -42,6 +42,9 @@ server name validation against the kubelet certificate.
 | `private_key_path` | The path on disk to client key used for kubelet authentication |
 | `node_name_env` | The environment variable used to obtain the node name. Defaults to `MY_NODE_NAME`. |
 | `node_name` | The name of the node. Overrides the value obtained by the environment variable specified by `node_name_env`. |
+| `skip_signature_verification_image_list`| The list of images, described as digest hashes, that should be skipped in signature verification. |
+| `enable_allowed_subjects_list`| Enables a list of allowed subjects that are trusted and are allowed to sign container images artificats.|
+| `allowed_subjects_list`| The list of allowed subjects enabled by `enable_allowed_subjects_list` each entry represents subject e-mail. |
```
We missed the Rekor URL here.

tjamir commented on doc/plugin_agent_workloadattestor_k8s.md:
```diff
@@ -117,7 +121,3 @@ WorkloadAttestor "k8s" {
-
-### Platform support
-
-This plugin is only supported on Unix systems.
```
In the current state of the SPIRE docs, this text is still mantained, why should we revove it?
https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md

tjamir commented on support/k8s/k8s-workload-registrar/config_crd.go:
```diff
@@ -203,7 +205,7 @@ func (c *CRDMode) Run(ctx context.Context) error {
 			IdentityTemplate:      c.IdentityTemplate,
 			Context:               c.Context,
 			IdentityTemplateLabel: c.IdentityTemplateLabel,
-			DNSNameTemplates:      *c.DNSNameTemplates,
```
Why is this line removed?

tjamir reviewed (changes requested):
Some minor changes requested.
Good job!

matheusdefarias committed:
refactor: adjusted some documentation in the plugin_agent_workloadattestor_k8s.md file (#36)

Co-authored-by: Matheus Santos <mfcs@cesar.org.br>

matheusdefarias committed:
refactor: included the containerID in the selector descriptions and included some examples too (#37)

Co-authored-by: Matheus Santos <mfcs@cesar.org.br>

Commit:
fix: pr review adjustments

Commit:
fix: more adjustments from pr review

Commit:
fix: lint fix - receiver name s should be consistent with previous receiver name sigstore for sigstoreImpl

Commit:
fix: fix lint

willallves committed:
Fix lint bug

willallves committed:
Merge pull request #17 from matheusdefarias/merge-spire-into-hpe-review-adjustments

Adjustments from PR review
